### PR TITLE
SystemMonitor: Ignore processes with no threads

### PR DESCRIPTION
**Pull request by maclarsson**
"Zombie" processes with no live threads can trip up
the refresh of the process list.
This patch checks for such processes and removes them
from the list of shown processes.

This resolves issue #13982 .

### CODE REVIEW

**Comment by nico** [comment]
Isn't it better to show zombie processes as zombie processes? They're usually the result of a bug and they fill up the PID table. So it's better to make them visible.

**Review by circl-lastname — CHANGES_REQUESTED** [review]

File: Userland/Applications/SystemMonitor/ProcessModel.cpp [hunk 1]
```diff
@@ -505,17 +505,20 @@ void ProcessModel::update()
...
-    // FIXME: Also remove dead threads from processes
-    for (auto tid : tids_to_remove) {
-        m_threads.remove(tid);
-        for (size_t i = 0; i < m_processes.size(); ++i) {
-            auto& process = m_processes[i];
-            process.threads.remove_all_matching([&](auto const& thread) { return thread->current_state.tid == tid; });
-            if (process.threads.size() == 0) {
-                m_processes.remove(i);
-                --i;
+    // Remove dead threads from processes and "zombie" processes withoud active thread
```
[the remark this answers is not in the document]
Comment (maclarsson) [reply Userland/Applications/SystemMonitor/ProcessModel.cpp]: Thank you for the hint -> typo corrected and commit ammended.

**Comment by maclarsson** [comment]
> Isn't it better to show zombie processes as zombie processes? They're usually the result of a bug and they fill up the PID table. So it's better to make them visible.

This can be changed easily, if preferred. This would then be consistent to the `ps` command, although it is not possible to kill these processes.

Should I ammend this commit or create a new one?

**Comment by circl-lastname** [comment]
For future reference, always rebase instead of merging in these cases.

**Pull request closed without merging**